src/common/conditional_compilation/include/openvino/cc/pass/itt.hpp

```diff
@@ -35,9 +38,46 @@ OV_CC_DOMAINS(ov_pass);
 
 #    define RUN_ON_MODEL_SCOPE(region) MATCHER_SCOPE_(ov_pass, OV_PP_CAT(region, _run_on_model))
 
+#    define ADD_MATCHER_OBJ_1(obj, region, ...) obj->add_matcher<region>(__VA_ARGS__);
+#    define ADD_MATCHER_OBJ_0(obj, region, ...)
+#    define ADD_MATCHER(obj, region, ...)                                                   \
+        OV_PP_CAT(ADD_MATCHER_OBJ_, OV_CC_SCOPE_IS_ENABLED(OV_PP_CAT3(ov_pass, _, region))) \
+        (obj, region, __VA_ARGS__)
+
+#    define REGISTER_PASS_1(obj, region, ...) obj.register_pass<region>(__VA_ARGS__);
+#    define REGISTER_PASS_0(obj, region, ...)
+#    define PASS_DEFAULT(region)      OV_CC_SCOPE_IS_ENABLED(OV_PP_CAT3(ov_pass, _, region))
+#    define PASS_RUN_ON_MODEL(region) OV_CC_SCOPE_IS_ENABLED(OV_PP_CAT3(ov_pass, _, OV_PP_CAT(region, _run_on_model)))
+#    define PASS_RUN_ON_FUNCTION(region) \
+        OV_CC_SCOPE_IS_ENABLED(OV_PP_CAT3(ov_pass, _, OV_PP_CAT(region, _run_on_function)))
+#    define OR_000                         0
+#    define OR_001                         1
+#    define OR_010                         1
+#    define OR_011                         1
+#    define OR_100                         1
+#    define OR_101                         1
+#    define OR_110                         1
+#    define OR_111                         1
+#    define CAT_3_CONDITION(OP1, OP2, OP3) OV_PP_CAT(OR_, OV_PP_CAT3(OP1, OP2, OP3))
```

Comment:
Is there possible a simpler method to get the condition? such as use "or" operation directly considering that `OP1` `OP2` and `OP3` should have the actual value during compiling time.

Reply:
How  do you think use below code block to replace 8 OR_xxx macro?
```suggestion
#define OV_OR_ARG_PLACEHOLDER_ONE_1        1,
#define OV_OR_ARG_PLACEHOLDER_TWO_1        1,
#define OV_FIRST_ARG_GET(val, ...)         val
#define OV_OR_(arg1_or_junk, arg2_or_junk) OV_FIRST_ARG_GET(arg1_or_junk arg2_or_junk2 0)
#define OV_OR_2(x, y)                      OV_OR_(OV_PP_CAT(OV_OR_ARG_PLACEHOLDER_ONE_, x), OV_PP_CAT(OV_OR_ARG_PLACEHOLDER_TWO_, y))
#define OV_OR_3(x, y, z)                   OV_OR_2(OV_OR_2(x, y), z)
```

Reply:
Hi @riverlijunjie , I don't get your idea.

For example: OV_OR_3(0,0,1) ->
OV_OR_2(OV_OR_2(0, 0), 0)  ->
OV_OR_2(OV_OR_(OV_PP_CAT(OV_OR_ARG_PLACEHOLDER_, x), OV_PP_CAT(OV_OR_ARG_PLACEHOLDER_, y)), 0) ->
OV_OR_2(OV_OR_(OV_PP_CAT(1, x), OV_PP_CAT(1, y)), 0) ->
OV_OR_2(OV_OR_(OV_PP_CAT(1, 0), OV_PP_CAT(1, 0)), 0) ->
OV_OR_2(OV_OR_(10, 10), 0) ->
OV_OR_2(10, 0) ->
OV_OR_(OV_PP_CAT(1, 10), OV_PP_CAT(1, 0)) ->
OV_OR_(110, 10) ->
110
?

But I only want to get 1 or 0.

Reply:
Sorry, 
#define OV_OR_ARG_PLACEHOLDER_        1,
-->
#define OV_OR_ARG_PLACEHOLDER_1       1,
Usage:
OV_OR_3(PASS_DEFAULT(region), PASS_RUN_ON_MODEL(region), PASS_RUN_ON_FUNCTION(region))

Per your example:
OV_OR_3(0,0,0) ->
OV_OR_2(OV_OR_2(0, 0), 0) ->
OV_OR_2(OV_OR_(OV_PP_CAT(OV_OR_ARG_PLACEHOLDER_, 0), OV_PP_CAT(OV_OR_ARG_PLACEHOLDER_, 0)), 0) ->
OV_OR_2(OV_FIRST_ARG_GET(0), 0) -->
OV_OR_2(0, 0) -->
0